ui/src/dashboards/components/CellEditorOverlay.js

```diff
@@ -63,6 +73,111 @@ class CellEditorOverlay extends Component {
     }
   }
 
+  handleAddThreshold = () => {
+    const {colors} = this.state
+
+    if (colors.length <= MAX_THRESHOLDS) {
+      const randomColor = _.random(0, GAUGE_COLORS.length)
+
+      const maxValue = Number(
+        colors.find(color => color.type === COLOR_TYPE_MAX).value
+      )
+      const minValue = Number(
+        colors.find(color => color.type === COLOR_TYPE_MIN).value
+      )
+
+      const colorsValues = _.mapValues(colors, 'value')
+      let randomValue
+
+      do {
+        randomValue = String(_.round(_.random(minValue, maxValue, true), 2))
+      } while (_.includes(colorsValues, randomValue))
```

Comment:
Whoa! A do-while loop! Sweet! 👍

Reply:
A great solution for that problem... I had a similar one with color hash for series name, and I never solved it. Great idea moving forward.

BTW, I would prefer using the toString method instead of the String constructor, like this:
```js
randomValue = _.round(_.random(minValue, maxValue, true), 2).toString()
```

You could also use more modern ES6 syntax:
```js
randomValue = `${_.round(_.random(minValue, maxValue, true), 2)}`
```

Reply:
Ah nice, I'll roll with template string